internal/cri/server/container_start.go

```diff
@@ -118,6 +118,14 @@ func (c *criService) StartContainer(ctx context.Context, r *runtime.StartContain
 	if ociRuntime.Path != "" {
 		taskOpts = append(taskOpts, containerd.WithRuntimePath(ociRuntime.Path))
 	}
+
+	// append endpoint to the options so that task manager can get task api endpoint directly
+	endpoint := sandbox.Endpoint
+	if endpoint.Protocol != "" && endpoint.Address != "" {
```

Comment:
add a IsValid method for Endpoint, because it is a little strange that we say the endpoint is empty either Protocol or Address is empty, only both of them are empty can we say the endpoint is empty(maybe the version should also be 0).

but we can say the endpoint is valid only if both Protocol and Address is not empty.